src/psyclone/core/signature.py

```diff
@@ -106,14 +106,19 @@ def __str__(self):
         return "%".join(self._signature)
 
     # ------------------------------------------------------------------------
-    def to_fortran(self, component_indices):
-        '''Converts this signature back to a Fortran expression using the
-        provided component indices.
+    def to_language(self, component_indices, language_writer=None):
```

Comment:
Since `FortranWriter` is stateless, I think you can just have `language_writer=FortranWriter()` here and drop the `if xxx is None` below.

Reply:
I can't here, I am suffering from a circular import if I import the FortranWriter outside of the functions.

Reply:
That's a shame.